scripts/actions/remove-completed-batch.js

```diff
@@ -17,4 +29,65 @@ const saveRemainingBatches = async () => {
   );
 };
 
+/**
+ * Gets the fileUris from the deserializedBatchUids, takes the fileUris and fetches
+ * the contextUids, and then the contextUids are deleted for deserialized batches
+ * @param {Array} deserializedBatchUids
+ * @returns {Promise}
+ */
+const removePageContext = async (fileUris) => {
+  const accessToken = await getAccessToken();
+
+  const fileNames = fileUris.map((fileUri) => {
+    const filepath = fileUri.replace(`src/content/`, '');
+    const slug = filepath.replace(`.mdx`, '');
+    const contextUrl = new URL(slug, DOCS_SITE_URL);
+    return contextUrl.href;
+  });
+
+  const { items } = await vendorRequest({
+    method: 'GET',
+    endpoint: `https://api.smartling.com/context-api/v2/projects/${PROJECT_ID}/contexts`,
+    accessToken,
+  });
+
+  const contextUids = items
+    .filter((item) => fileNames.includes(item.name))
+    .map((item) => {
+      return {
+        fileUri: item.name,
+        contextUid: item.contextUid,
+      };
+    });
+
+  let returnCode = 'SUCCESS';
+
+  for (const { contextUid, fileUri } of contextUids) {
```

Comment:
As much as I love `for...of`, we might want to use `reduce` here snce we're looping over something (`contextUids`) and hoping to have one value at the end (`returnCode`).

Reply:
yeah I agree! I at first was not returning anything (which is why I was doing this) but I was running into some promise issues and returning a New Promise was causing a bunch of other errors.